src/Client/CurlHandler.php

```diff
@@ -57,8 +57,25 @@ public function __destruct()
         }
     }
 
+    /**
+     * @param array $request
+     *
+     * @return CompletedFutureArray
+     */
     public function __invoke(array $request)
     {
+        return new CompletedFutureArray(
+            $this->invokeAsArray($request)
+        );
+    }
+
+    /**
+     * @param array $request
+     *
+     * @return array
+     */
+    public function invokeAsArray(array $request)
```

Comment:
No that is not possible. That function will be given as the callback ($handle in CurlFactory)
I do agree it is weird.

Do you have any thoughts?